System/Random.hs

```diff
@@ -949,91 +952,90 @@ runSTGen :: RandomGen g => g -> (forall s . STGen g s -> ST s a) -> (a, g)
 runSTGen g action = unSTGen <$> runST (runGenM (STGen g) action)
 
 -- | Runs a monadic generating action in the `ST` monad using a pure
--- pseudo-random number generator. Returns only the resulting random value.
+-- pseudo-random number generator. Returns only the resulting pseudo-random
+-- value.
 --
 -- @since 1.2
 runSTGen_ :: RandomGen g => g -> (forall s . STGen g s -> ST s a) -> a
 runSTGen_ g action = fst $ runSTGen g action
 
--- | The default pseudo-random number generator.
-type StdGen = SM.SMGen
+-- | The standard pseudo-random number generator. Uses the SplitMix
+-- implementation provided by the
+-- [splitmix](https://hackage.haskell.org/package/splitmix) package.
+newtype StdGen = StdGen { unStdGen :: SM.SMGen }
```

Comment:
Is there a reason why we are hiding the access to `SMGen`? I understand desire to wrap it with a `newtype`, but making it private is too restrictive

Reply:
My thinking was that this allows for forward-compatibility. As per our reproducibility guarantee, changing the underlying PRNG would require a major version bump. But this way we can change the underlying PRNG without also breaking the API.

What would be the advantages of making `StdGen`'s public, e.g. via `type StdGen = SMGen`?

Reply:
I think changing the algorithm would be a breaking change too regardless, because properties of numbers being generated purely would change. Also same `Int` seed would result in different values being generated.

If someone wants to use `SMGen` directly they would be able to do so:
> What would be the advantages of making StdGen's public, e.g. via type StdGen = SMGen?

My point is, it is not always good to completely hide the implementation from the user. I am thinking maybe we can introduce an Internal module, where all of the internal stuff can be exported from?

Reply:
> we can introduce an Internal module, where all of the internal stuff can be exported from?

That sounds like a good idea to me. This should happen in a separate PR though. I've reverted the change to `StdGen` for now.

Reply:
Cool. If wrapping with the newtype is the way to go, then we are also missing an `instance RandomGen SM.SMGen`

Reply:
Oh and then instance for `StdGen` can be derived with `GeneralizedNewtypeDeriving`